resolve.c

```diff
@@ -88,16 +88,36 @@ void resolveClients(bool onlynew)
 		// Lock data when obtaining IP of this client
 		lock_shm();
 		const char* ipaddr = getstr(clients[clientID].ippos);
+		const char* name = getstr(clients[clientID].namepos);
 		unlock_shm();
 
 		// Important: Don't hold a lock while resolving as the main thread
 		// (dnsmasq) needs to be operable during the call to resolveHostname()
-		const char* hostname = resolveHostname(ipaddr);
+		char* hostname = resolveHostname(ipaddr);
 
 		// Finally, lock data when storing obtained hostname
 		lock_shm();
-		clients[clientID].namepos = addstr(hostname);
+
+		// Only store new hostname if it changed
+		if(hostname != NULL && strcmp(name, hostname) != 0)
+		{
+			// We do not need to check for name == NULL as name is
+			// always initialized with an empty string at position 0
+			clients[clientID].namepos = addstr(hostname);
+		}
+		else if(config.debug & DEBUG_SHMEM)
+		{
+			// Debugging output
+			logg("Not adding \"%s\" to buffer (unchanged)", name);
+		}
+
+		// Release allocated memory
+		free(hostname);
```

Comment:
The hostname could be null, so this would try to free a null pointer.

See line 67.

Reply:
Our own implementation of `free()` is robust against trying to free `NULL` and will only log a warning into FTL's log. However, I agree that this should be fixed and did this now.